pkg/services/publicdashboards/service/service.go

```diff
@@ -238,6 +246,90 @@ func (pd *PublicDashboardServiceImpl) GetMetricRequest(ctx context.Context, dash
 	return metricReqDTO, nil
 }
 
+func (pd *PublicDashboardServiceImpl) GetAnnotations(ctx context.Context, reqDTO AnnotationsQueryDTO, accessToken string) ([]AnnotationEvent, error) {
+	_, dash, err := pd.GetPublicDashboard(ctx, accessToken)
+	if err != nil {
+		return nil, err
+	}
+
+	type annotationsDto struct {
+		Annotations struct {
+			List []DashAnnotation `json:"list"`
+		}
+	}
+
+	dto := annotationsDto{}
+	bytes, err := dash.Data.MarshalJSON()
+	if err != nil {
+		return nil, err
+	}
+	err = json.Unmarshal(bytes, &dto)
+	if err != nil {
+		return nil, err
+	}
+
+	anonymousUser, err := pd.BuildAnonymousUser(ctx, dash)
+	if err != nil {
+		return nil, err
+	}
+
+	uniqueEvents := make(map[int64]AnnotationEvent, 0)
+	for _, anno := range dto.Annotations.List {
+		// only consider enabled, grafana annotations
+		if anno.Enable && (*anno.Datasource.Uid == grafanads.DatasourceUID || *anno.Datasource.Uid == grafanads.DatasourceName) {
```

Comment:
I prefer this to avoid nesting and improve legibility
```suggestion
		if anno.Enable && (*anno.Datasource.Uid == grafanads.DatasourceUID || *anno.Datasource.Uid == grafanads.DatasourceName) {
		continue
		}
```